int32 count
string str
